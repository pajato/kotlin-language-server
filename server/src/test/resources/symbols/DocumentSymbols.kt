private class DocumentSymbols() {
    val aProperty = 1

    fun aFunction(aFunctionArg: Int) {
        println("Hello Kotlin Language Server")
    }

    constructor(aConstructorArg: Int): this() {
    }
}
