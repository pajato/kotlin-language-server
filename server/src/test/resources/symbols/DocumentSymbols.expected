DocumentSymbol("||path-to-symbols||/DocumentSymbols.kt", File, KtFile, private class DocumentSymbols() {@    val aProperty = 1@@    fun aFunction(aFunctionArg: Int) {@        println("Hello Kotlin Language Server")@    }@@    constructor(aConstructorArg: Int): this() {@    }@}@)
    DocumentSymbol("", Package, KtPackageDirective, )
    DocumentSymbol("DocumentSymbols", Class, KtClass, private class DocumentSymbols() {@    val aProperty = 1@@    fun aFunction(aFunctionArg: Int) {@        println("Hello Kotlin Language Server")@    }@@    constructor(aConstructorArg: Int): this() {@    }@})
        DocumentSymbol("DocumentSymbols", Constructor, KtPrimaryConstructor, ())
        DocumentSymbol("", Class, KtClassBody, {@    val aProperty = 1@@    fun aFunction(aFunctionArg: Int) {@        println("Hello Kotlin Language Server")@    }@@    constructor(aConstructorArg: Int): this() {@    }@})
            DocumentSymbol("aProperty", Property, KtProperty, val aProperty = 1)
            DocumentSymbol("aFunction", Function, KtNamedFunction, fun aFunction(aFunctionArg: Int) {@        println("Hello Kotlin Language Server")@    })
                DocumentSymbol("(aFunctionArg: Int)", Function, KtParameterList, (aFunctionArg: Int))
                DocumentSymbol("aFunction", Function, KtBlockExpression, {@        println("Hello Kotlin Language Server")@    })
                    DocumentSymbol("aFunction", Function, KtCallExpression, println("Hello Kotlin Language Server"))
                        DocumentSymbol("println", Function, KtNameReferenceExpression, println)
                        DocumentSymbol("aFunction", Function, KtValueArgumentList, ("Hello Kotlin Language Server"))
            DocumentSymbol("DocumentSymbols", Constructor, KtSecondaryConstructor, constructor(aConstructorArg: Int): this() {@    })
                DocumentSymbol("DocumentSymbols", Constructor, KtConstructorDelegationCall, this())
                    DocumentSymbol("DocumentSymbols", Constructor, KtValueArgumentList, ())
                DocumentSymbol("DocumentSymbols", Constructor, KtBlockExpression, {@    })
